src/py/crankshaft/crankshaft/analysis_data_provider.py

```diff
@@ -2,18 +2,25 @@
 import plpy
 import pysal_utils as pu
 
+NULL_VALUE_ERROR = ('No usable data passed to analysis. Check your input rows '
+                    'for null values and fill in appropriately.')
+
+
+def verify_data(n_rows):
```

Comment:
Since the data exchange is 100% in Python, it shouldn't make a difference -- you're just passing around a reference to the array, none of this could should result in a copy of the data ever being made.